src/Features/Core/Portable/ConvertAnonymousTypeToClass/AbstractConvertAnonymousTypeToClassCodeRefactoringProvider.cs

```diff
@@ -70,34 +70,13 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
         private async Task<(TAnonymousObjectCreationExpressionSyntax, INamedTypeSymbol)> TryGetAnonymousObjectAsync(
             Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            var position = span.Start;
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(position);
-
-            // Span actually has to be within the token (i.e. not in trivia around it).
-            if (!token.Span.IntersectsWith(position))
-            {
-                return default;
-            }
-
-            if (!span.IsEmpty && span != token.Span)
-            {
-                // if there is a selection, it has to be of the whole token.
-                return default;
-            }
-
-            var anonymousObject = token.Parent as TAnonymousObjectCreationExpressionSyntax;
+            var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
+            var anonymousObject = await refactoringHelperService.TryGetSelectedNodeAsync<TAnonymousObjectCreationExpressionSyntax>(document, span, cancellationToken).ConfigureAwait(false);
             if (anonymousObject == null)
             {
                 return default;
             }
 
-            // The position/selection must be of the 'new' token of the anonymous object.
-            if (anonymousObject.GetFirstToken() != token)
```

Comment:
This is potentially a breaking change that (might) be undesirable. With this we start offering the refactoring for few more caret positions & selections e.g:

Due to `{`, `,` being tokens just under AnonymousObjectCreationSyntax

```
        var q = new [||]{ m="a"[||], n="b" };
```
Due to being lenient when it comes to partial selections:
```
       var q = new {[||] m="a", [||]n="b" };
```
`

Reply:
I'm not exactly sure how to solve it in a nice way but we could simply test that the start of selection / caret is within/around the new keyboard. Maybe add a TypeParam to the helper that caret / selection must start/include `TToken` token?

Reply:
having hte refactorin offered on those cases seems ok to me.

Reply:
Ok, will add comment warning about it.